packages/image-cloze-association/controller/src/index.js

```diff
@@ -0,0 +1,96 @@
+import debug from 'debug';
+import _ from 'lodash';
+import { camelizeKeys } from 'humps';
+
+const log = debug('pie-elements:image-cloze-association:controller');
+
+import { getNumberOfDuplicates } from './utils';
+
+export function model(question, session, env) {
+  const questionCamelized = camelizeKeys(question);
+
+  return new Promise(resolve => {
+    const out = {
+      disabled: env.mode !== 'gather',
+      mode: env.mode,
+      ...questionCamelized,
+      responseCorrect:
+        env.mode === 'evaluate'
+          ? isResponseCorrect(questionCamelized, session)
+          : undefined,
+    };
+
+    resolve(out);
+  });
+}
+
+const isResponseCorrect = (question, session) => {
+  const { validation: { validResponse } } = question;
+  let isCorrect = true;
+  let totalValidResponses = 0;
+
+  validResponse.value.forEach(value => totalValidResponses += value.length);
+
+  if (session.answers && totalValidResponses === session.answers.length) {
+    session.answers.forEach(answer => {
+      if (!validResponse.value[answer.containerIndex].includes(answer.value)) {
+        isCorrect = false;
+      }
+    });
+  } else {
+    isCorrect = false;
+  }
+  return isCorrect;
+};
+
+const isResponsePartiallyCorrect = (question, session) => {
```

Comment:
naming a function `is****` implies you are going to return a `boolean` but you return a float. so maybe rename to what that float means? like `getPartialScore`?